hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/HighlightingFormatter.java

```diff
@@ -0,0 +1,30 @@
+package org.hibernate.engine.jdbc.internal;
+
+import java.util.regex.Pattern;
+
+/**
+ * Performs basic syntax highlighting for SQL using ANSI escape codes.
+ *
+ * @author Gavin King
+ */
+public class HighlightingFormatter implements Formatter {
+
+    private static final Pattern keywords =
+            Pattern.compile("\\b(create|drop|alter|add|table|sequence|primary|foreign|unique|key|constraint|cascade|references|not|null|if|exists|select|as|delete|insert|update|set|from|in|into|values|left|right|inner|outer|join|on|where|having|order|group|by|desc|asc)\\b");
+    private static final Pattern strings =
+            Pattern.compile("'[^']*'");
```

Comment:
Guess we don't need to make everything perfect, but RE based matching might be broken for the following SQL:
```
update book b set b.title = 'Bob Dylan\'s dream' where b.id = 1;
```

Reply:
Yeah look, I'm not sure if it's really worth trying to make it completely perfect, but in principle I could ultimately get rid of the regexes and write a proper tokenizer for SQL. That's not hard.

Reply:
Oh and also the correct way to escape a ' in a SQL string is '', isn't it?

Reply:
Yeah. It is double single quotes.

Reply:
Yeah, if it is really worth trying to make it perfect, we might well get formatting info from SQL generation phase, which might be more robust than RE. Anyway, if something imperfect is released, we might need to fix some bug reported by our users sooner or later.

Reply:
I've changed it to use a handwritten tokenizer.